orangecontrib/bioinformatics/annotation/annotate_samples.py

```diff
@@ -0,0 +1,174 @@
+from collections import defaultdict
+
+import numpy as np
+from Orange.data import Domain, ContinuousVariable, Table
+
+from orangecontrib.bioinformatics.utils import statistics
+from scipy.stats.mstats import rankdata
+from scipy.stats import hypergeom
+
+
+class AnnotateSamples:
+    """
+    AnnotateSamples is class used for the annotation of data items with the
+    labels Mann-Whitney U test for selecting important values and
+    the Hyper-geometric for assigning the labels.
+
+    Example of use:
+
+    >>> from Orange.data import Table
+    >>> from orangecontrib.bioinformatics.utils import serverfiles
+    >>> from orangecontrib.bioinformatics.annotation.annotate_samples import AnnotateSamples
+    >>>
+    >>> data = Table("https://datasets.orange.biolab.si/sc/aml-1k.tab.gz")
+    >>> markers_path = serverfiles.localpath_download(
+    >>>     'marker_genes','panglao_gene_markers.tab')
+    >>> marker = Table(markers_path)
+    >>> annotator = AnnotateSamples(p_value_th=0.05)
+    >>> annotations = annotator.annotate_samples(data, markers)
```

Comment:
Only Human markers should go into this example. In reality this will be guaranteed by the Marker Genes widget.

```
from Orange.data.filter import FilterString, Values
f = FilterString("Organism", FilterString.Equal, organism)
g = FilterString("Entrez ID", FilterString.NotEqual, "?")
markers = Values([f, g])(markers)
```